specs/clipboard-api/clipboard-api.md

```diff
@@ -0,0 +1,337 @@
+  if (dataPackageView.Contains(StandardDataFormats.Text))
+  {
+    string text = await dataPackageView.GetTextAsync();
+    // To output the text from this example, you need a TextBlock control
+    TextOutput.Text = "Clipboard now contains: " + text;
+  }
+}
+```
+
+## Remarks
+
+\[None at this time\]
+
+## API Notes
+
+The syntactic interface of the Project Reunion SDK's clipboard API
+is a subset of [Windows.ApplicationModel.DataTransfer](
+https://docs.microsoft.com/uwp/api/windows.applicationmodel.datatransfer)
+with a different name (Microsoft.ProjectReunion.ApplicationModel).
+Certain types (classes and enums) are omitted, and one class, `DataPackage`,
+omits 2 members that are not relevant to clipboard usage.
+
+The set of types from Windows.ApplicationModel.DataTransfer that
+will be copied into the clipboard API is the following:
+
+* [`Clipboard`](https://docs.microsoft.com/uwp/api/windows.applicationmodel.datatransfer.clipboard) -
+  Top level for the clipboard API. Provides "static" methods
+  (methods implemented by the class's activation factory) to read and write
+  the current clipboard item and clipboard history, along with methods
+  to read settings for clipboard history and cloud clipboard sync.
+
+* `Clipboard`-related types:
+  * `ClipboardContentOptions`
+  * `ClipboardHistoryItem`
+  * `ClipboardHistoryItemsResult`
+  * `ClipboardHistoryItemsResultStatus`
+  * `OperationCompletedEventArgs`
+  * `SetHistoryItemAsContentStatus`
+  * (`ClipboardHistoryChangedEventArgs` is omitted because it's an empty class.)
+
+* [`DataPackage`](https://docs.microsoft.com/uwp/api/windows.applicationmodel.datatransfer.datapackage) -
+  Unit of data transfer for the clipboard API.
+  The `ShareCompleted` and `ShareCanceled` events are omitted.
+
+* `DataPackage`-related types:
+  * `DataPackageView`
+  * `DataPackageOperation`
+  * `DataPackagePropertySet`
+  * `DataPackagePropertySetView`
+  * `DataProviderDeferral`
+  * `DataProviderHandler`
+  * `DataProviderRequest`
+  * `HtmlFormatHelper`
+  * `StandardDataFormats`
+
+## API Details
+
+See [`clipboard.idl`](clipboard.idl) for a complete [MIDL 3.0](
+https://docs.microsoft.com/uwp/midl-3/) description
+of the Project Reunion SDK's clipboard API.
+
+## Appendix
+
+Initially, the implementation of the Project Reunion SDK's clipboard APi
+will be a set of very thin wrappers around the same-named codes in
+[Windows.ApplicationModel.DataTransfer](
+https://docs.microsoft.com/uwp/api/windows.applicationmodel.datatransfer),
+which will gracefully fail with E_NOTIMPL or similar for features that
+are not implemented on that Windows version. (This will be mostly limited
+to cloud clipboard roaming and clipboard history.)
+
+The only exception will be the change to make [Clipboard.Flush()](
+https://docs.microsoft.com/uwp/api/windows.applicationmodel.datatransfer.clipboard.flush)
+a completely optional operation.
+(TO CONSIDER: Should Clipboard.Flush() be removed entirely?)

```

Comment:
As discussed in an internal thread, let's be careful with removing this. Flushing by default would be
a) absolute opposite of what OLE clipboard layer does (delay render by default)
b) make clipboard change operation potentially very expensive for the apps, and most likely wasteful, since there might not be any consumers for some of the formats that app is advertising

Reply:
For clarity’s sake: When I recommended getting rid of `Clipboard.Flush()` I was mainly talking about the public method, to avoid confusion. @andrew-z raises a good point re performance; I did not think of that. I agree with @metathinker's idea about providing an auto-flushing opt out.